### PR TITLE
ks upgrade relocates vendored packages to versioned paths

### LINKED ISSUE #663: ks upgrade moves vendored packages to versioned paths
ks upgrade should relocate previously (ks < 0.12.0) vendored paths to their new, versioned locations ( >= 0.12.0).

Example:

`vendor/<registry>/<pkg>` -> `vendor/<registry>/<pkg>@<version>`

Part of #629

### PR DESCRIPTION
**Pull request by shomron**
Example:
`vendor/<registry>/<pkg>` -> `vendor/<registry>/<pkg>@<version>`

Closes #663

Signed-off-by: Oren Shomron <shomron@gmail.com>

### CODE REVIEW

**Comment by bryanl** [comment]
I didn't notice this before, but prototypes don't show up if you haven't upgraded your app to 0.2.0.

**Comment by shomron** [comment]
^ @bryanl that is no good. Thanks for surfacing it!

File: pkg/upgrade/upgrade010.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+	"github.com/ksonnet/ksonnet/pkg/app"
+	"github.com/ksonnet/ksonnet/pkg/lib"
+	"github.com/ksonnet/ksonnet/pkg/pkg"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+)
+
+// Package is a subset of the pkg.Package interface for use in PackageLister
+// type Package interface {
```
Comment (bryanl) [inline pkg/upgrade/upgrade010.go]: This isn't used

File: pkg/upgrade/upgrade010.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+}
+
+var removeVersionPattern = regexp.MustCompile("(.*)@.*$")
+
+// In ksonnet 0.12.0 (app version 0.2.0) - the vendor cache began storing
+// packages using versioned paths. This upgrade translates old paths into new, versioned paths.
+// Example:
+// `vendor/incubator/mysql` -> `vendor/incubator/mysql@1.2.3`
+func (u *upgrade010) upgradeOldVendoredPackages(pl PackageLister, dryRun bool) error {
+	// log := logrus.WithFields(logrus.Fields{
```
Comment (bryanl) [inline pkg/upgrade/upgrade010.go]: This isn't used

**Comment by shomron** [comment]
Opened #684

**Comment by shomron** [comment]
Removed dead code.

**Pull request merged**